### DISCUSSION TITLE
posts/code-server-self-host/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#272] [Comments]
# posts/code-server-self-host/

Have you ever wanted to run VS Code in your browser? What if you had access to your terminal and could pull and commit code as well as push it up to GitHub all from a browser or tablet? That’s exactly what code server does! In this tutorial we’ll walk through step by step of how to install and configure code server to get it self-hosted in your homelab.We’ll start with bare metal and virtualization and then work our way up to Docker, Kubernetes, and Rancher.Then, you don’t have to carry around your laptop anymore! You can preserve battery life on the go and leave the intensive tasks to your homelab server.

https://technotim.live/posts/code-server-self-host/

**Comment by emiellr** [comment]
Ok but if I run code-server in Docker, (how) would I be able to compile with GCC or whatever? Wouldn't a recreation of the Docker image just wipe GCC again?

**Comment by AxiomaAbsurdo** [comment]
Hi Tim. 🙋‍♂️
I am going to take advantage of the recent update (February 22, 2024) of this beautiful article to ask some questions.
You may know AWS workstations, they are virtual machines as powerful as physical ones, and as or more expensive than physical ones. 💸💸💸

Since the middle of last year, I have set up a decent home lab that I believe is powerful enough to serve as an alternative to AWS Workspaces for many tasks. My goal is to create Proxmox containers configured as complete development environments that I can build and customize as needed. For example, I'd like to have containers set up for web development with TypeScript, Android development with Flutter/Dart, and Python development with Arduino.

I'm wondering if you know of any guides, resources, or examples of people who have set up similar containerized development environments on home labs and made them accessible remotely like AWS Workspaces. Specifically, I'm looking for information on:

 - Setting up the dev environment containers in Proxmox
 - Enabling remote access via methods like WireGuard
 - Adding an access/authentication manager like OKTA or an open source alternative to Auth0

I appreciate any pointers you can provide. Please let me know if any part of this is unclear! I'm aiming to replicate the convenience of AWS Workspaces using my own home lab.

Thank you.🙏
